src/EnumBitMask.php

```diff
@@ -19,16 +21,27 @@ final class EnumBitMask implements BitMaskInterface
 
     /**
      * @param class-string $enum
+     * @throws InvalidEnumException
      * @throws UnknownEnumException
      */
     public function __construct(
         private readonly string $enum,
         int $mask = 0,
+        bool $isIntBacked = false,
```

Comment:
I noticed we instantiate the class in static functions. (like the all() method for example). We might run into consistency issues?  It might be worth considering to treat all int backed enums as "isIntBacked". But that change would not be backward compatible, as people might be using int backed enum with the existing implementation.

That's why I was thinking about a subclass like IntBackedEnumBitmask.

Reply:
Thank you for your thoughts. You’re right that treating all int-backed enums as "isIntBacked" could lead to consistency issues and wouldn't be backward compatible. That's why the current approach includes the boolean parameter to ensure flexibility. However, creating a subclass like IntBackedEnumBitmask could be a cleaner solution in the long term.